### PR TITLE
fix TypeError: pair[1] is undefined when query param doesn't have = sign

### PR DESCRIPTION
**Pull request by baso10**
TypeError: pair[1] is undefined

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

**Comment by baso10** [comment]
/?model[field]&model[field2]=123 will disable searching. (Going into column filter input and press enter will fail with exception TypeError: pair[1] is undefined

GridView -> getClientOptions() populates "filterUrl". This is then parsed in yii.gridView.js and passed on yii.js, where code expects "=" sign.

To reproduce you can manually remove "=" from browser url bar and press enter.

File: framework/assets/yii.js [hunk 1]
```diff
@@ -297,7 +297,7 @@ window.yii = (function ($) {
             for (var i = 0, len = pairs.length; i < len; i++) {
                 var pair = pairs[i].split('=');
                 var name = decodeURIComponent(pair[0].replace(/\+/g, '%20'));
-                var value = decodeURIComponent(pair[1].replace(/\+/g, '%20'));
+                var value = pair.length > 1 ? decodeURIComponent(pair[1].replace(/\+/g, '%20')) : '';
```
Comment (samdark) [inline framework/assets/yii.js]: Did you mean > 0?

Comment (baso10) [reply framework/assets/yii.js]: No. It needs to be length 2, so pair[1] is not null

**Comment by baso10** [comment]
Any update?

**Comment by samdark** [comment]
No. Was not able to verify it. A test may speed things up with the review/merge.

**Comment by baso10** [comment]
Any update? Test is done

**Review by samdark — APPROVED** [review]

**Review by samdark — APPROVED** [review]
Looks good for the next release. Please add a line for CHANGELOG.md file.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you 👍

**Pull request merged**